### DISCUSSION TITLE
Securitytools Module Enhancement

### DISCUSSION THREAD

**Discussion by Sh4d0wHunt3rX** [blacklanternsecurity/bbot#1581] [General]
That would be great if possible to add these apis to securitytrails module:

https://docs.securitytrails.com/reference/history-dns
https://docs.securitytrails.com/reference/company-associated-ips
https://docs.securitytrails.com/reference/domain-associated-domains
https://docs.securitytrails.com/reference/domain-ssl-certificates-stream

Also, subfinder is using this one, not sure if needed:
https://docs.securitytrails.com/reference/scroll

Thanks 🙏

**Comment by Sh4d0wHunt3rX** [comment]
@TheTechromancer  I was checking the Securitytrails module closely and I noticed you used:

```
url = f"{self.base_url}/domain/{query}/subdomains?apikey={self.api_key}"
```
![image](https://github.com/user-attachments/assets/a6a029cf-d4b6-428d-a54c-51c0d3a19c30)

![image](https://github.com/user-attachments/assets/5630eee7-29f3-4b90-9ca2-37dbc8029f40)

For example security trails has 53920 subdomains for dell.com

![image](https://github.com/user-attachments/assets/8454343b-14bf-4b72-8a83-a679a6e3eee3)

That's why subfinder is also using:
```
resp, err = session.Post(ctx, "https://api.securitytrails.com/v1/domains/list?include_ips=false&scroll=true",
```

**Reply by TheTechromancer** [reply]
Rgr, that's a good suggestion. I'll make an issue.
